compiler/ir/backend.js/src/org/jetbrains/kotlin/ir/backend/js/lower/AutoboxingTransformer.kt

```diff
@@ -113,20 +160,34 @@ class AutoboxingTransformer(val context: JsIrBackendContext) : AbstractValueUsag
         }
     }
 
+    private tailrec fun IrExpression.isGetUnit(): Boolean =
+        when (this) {
+            is IrContainerExpression ->
+                when (val lastStmt = this.statements.lastOrNull()) {
+                    is IrExpression -> lastStmt.isGetUnit()
+                    else -> false
+                }
+
+            is IrGetObjectValue ->
+                this.symbol == irBuiltIns.unitClass
+
+            else -> false
+        }
+
     private fun buildSafeCall(
         arg: IrExpression,
         actualType: IrType,
         resultType: IrType,
         call: (IrExpression) -> IrExpression
     ): IrExpression {
-        if (!actualType.isNullable())
+        if (!actualType.isNullable() || !resultType.isNullable())
             return call(arg)
         return JsIrBuilder.run {
             // TODO: Set parent of local variables
             val tmp = buildVar(actualType, parent = null, initializer = arg)
             val nullCheck = buildIfElse(
                 type = resultType,
-                cond = buildCall(irBuiltIns.eqeqSymbol).apply {
+                cond = buildCall(irBuiltIns.eqeqeqSymbol).apply {
```

Comment:
`tmp === null` will be false from `undefined`, couldn't it broke something?

Reply:
Reverted it. It was needed before I implemented "== null" optimization in Wasm's IR intrinsic lowering.